core/src/main/java/io/questdb/griffin/SqlCodeGenerator.java

```diff
@@ -5128,16 +5128,17 @@ private RecordCursorFactory generateTableQuery(
     ) throws SqlException {
         final ObjList<ExpressionNode> latestBy = model.getLatestBy();
 
-        final BufferWindowCharSequence tab = (BufferWindowCharSequence) model.getTableName();
         final boolean supportsRandomAccess;
-        if (Chars.startsWith(tab, NO_ROWID_MARKER)) {
+        CharSequence tableName = model.getTableName();
+        if (Chars.startsWith(tableName, NO_ROWID_MARKER)) {
```

Comment:
why this change was needed?

Reply:
The `(BufferWindowCharSequence) tableName` downcast started failing once I added the pseudo-function `query_trace()` through AST rewriting. It replaces the token `query_trace()` with `sys.query_trace`.